CS3219-AY2223S1/cs3219-project-ay2223s1-g12
# Implement storage of returned JWT in cookie for successful logins

Fixes #62 

**Summary**
* Successful logins return jwt tokens but they are not saved in a cookie
* Storage of jwt as a cookie is necessary for authorization requests
* In this PR, jwt token is stored as a _httpOnly_ cookie to prevent javascript from accessing it
* Additional notes: `app.use(cors())` in user-service/index.js has to be disabled for this feature to work. When we do `app.use(cors())`, all of the requests are allowed by default and because of which, the 'Access-Control-Allow-Origin' header is set to '*'. This leads to CORS issues as the front-end and user-service are on different origins. The conflict arising from '*' usage is due to the browser requiring additional credential arguments in the axios.post() call, otherwise it would not save cookies, and these additional credential arguments do not accept '*' in the 'Access-Control-Allow-Origin' header.

Steps to test:
1. Navigate to `http://localhost:3000/login` and proceed to sign in.
2. Upon redirection to home page, open browser's developer options > Application tab. Observe that a cookie 'token' is saved on the browser.
